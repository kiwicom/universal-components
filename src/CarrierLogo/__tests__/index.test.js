// @flow

import * as React from 'react';
import { Image } from 'react-native';

import { render } from 'react-native-testing-library';
import snapshotDiff from 'snapshot-diff';
import { CarrierLogo } from '..';

describe('CarrierLogo', () => {
  const carriers = [
    { code: 'FR', name: 'Ryanair' },
    { code: 'TO', name: 'Transavia France' },
    { code: 'VY', name: 'Vueling' },
    { code: 'OK', name: 'Czech Airlines' },
  ];
  const { getAllByType } = render(<CarrierLogo carriers={carriers} />);

  it('should contain correct number of images', () => {
    expect(getAllByType(Image)).toHaveLength(4);
  });

  it('should match snapshot diff', () => {
    const base = <CarrierLogo carriers={[carriers[0]]} />;
    const extend = <CarrierLogo carriers={carriers} />;
    expect(snapshotDiff(base, extend)).toMatchSnapshot();
  });
});
