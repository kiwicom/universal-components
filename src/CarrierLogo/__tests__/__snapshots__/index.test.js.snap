// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarrierLogo should match snapshot diff 1`] = `
"Snapshot Diff:
- <CarrierLogo carriers={[{\\"code\\": \\"FR\\", \\"name\\": \\"Ryanair\\"}]} />
+ <CarrierLogo carriers={[{\\"code\\": \\"FR\\", \\"name\\": \\"Ryanair\\"}, {\\"code\\": \\"TO\\", \\"name\\": \\"Transavia France\\"}, {\\"code\\": \\"VY\\", \\"name\\": \\"Vueling\\"}, {\\"code\\": \\"OK\\", \\"name\\": \\"Czech Airlines\\"}]} />

@@ -7,11 +7,11 @@
          \\"display\\": \\"flex\\",
          \\"flexWrap\\": \\"wrap\\",
          \\"justifyContent\\": \\"space-between\\",
        },
        Object {
-         \\"flexDirection\\": \\"row\\",
+         \\"flexDirection\\": \\"column\\",
        },
        Object {
          \\"height\\": 32,
          \\"width\\": 32,
        },
@@ -19,23 +19,83 @@
    }
  >
    <Image
      source={
        Object {
-         \\"uri\\": \\"https://images.kiwi.com/airlines/64/FR.png?default=airline.png\\",
+         \\"uri\\": \\"https://images.kiwi.com/airlines/32/FR.png?default=airline.png\\",
        }
      }
      style={
        Array [
          Object {
            \\"backgroundColor\\": \\"transparent\\",
            \\"borderRadius\\": 3,
          },
          Object {
-           \\"height\\": 32,
-           \\"width\\": 32,
+           \\"height\\": 15,
+           \\"width\\": 15,
          },
          false,
        ]
      }
    />
+   <Image
+     source={
+       Object {
+         \\"uri\\": \\"https://images.kiwi.com/airlines/32/TO.png?default=airline.png\\",
+       }
+     }
+     style={
+       Array [
+         Object {
+           \\"backgroundColor\\": \\"transparent\\",
+           \\"borderRadius\\": 3,
+         },
+         Object {
+           \\"height\\": 15,
+           \\"width\\": 15,
+         },
+         false,
+       ]
+     }
+   />
+   <Image
+     source={
+       Object {
+         \\"uri\\": \\"https://images.kiwi.com/airlines/32/VY.png?default=airline.png\\",
+       }
+     }
+     style={
+       Array [
+         Object {
+           \\"backgroundColor\\": \\"transparent\\",
+           \\"borderRadius\\": 3,
+         },
+         Object {
+           \\"height\\": 15,
+           \\"width\\": 15,
+         },
+         false,
+       ]
+     }
+   />
+   <Image
+     source={
+       Object {
+         \\"uri\\": \\"https://images.kiwi.com/airlines/32/OK.png?default=airline.png\\",
+       }
+     }
+     style={
+       Array [
+         Object {
+           \\"backgroundColor\\": \\"transparent\\",
+           \\"borderRadius\\": 3,
+         },
+         Object {
+           \\"height\\": 15,
+           \\"width\\": 15,
+         },
+         false,
+       ]
+     }
+   />
  </View>"
`;
